data-prepper-core/src/main/java/org/opensearch/dataprepper/core/pipeline/server/UpdatePipelineBaseHandler.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.core.pipeline.server;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import org.opensearch.dataprepper.core.pipeline.PipelinesProvider;
+import org.opensearch.dataprepper.model.configuration.PipelinesDataFlowModel;
+import org.opensearch.dataprepper.pipeline.parser.PipelineConfigurationS3FileReader;
+import org.opensearch.dataprepper.pipeline.parser.PipelinesDataflowModelParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.regions.Region;
+
+import javax.ws.rs.HttpMethod;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * HttpHandler to handle requests for updating pipeline configurations from S3
+ */
+public abstract class UpdatePipelineBaseHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UpdatePipelineBaseHandler.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Pattern PIPELINE_NAME_PATTERN = Pattern.compile("/([a-zA-Z0-9-]{1,28})$");
```

Comment:
This regex appears overly restrictive. I don't think we have a current pipeline name requirement actually.